src/main/java/bean/parser/Parser.java

```diff
@@ -0,0 +1,45 @@
+package bean.parser;
+
+import bean.command.Command;
+import bean.exception.UnknownCommandException;
+
+/**
+ * Parses user input and generates the appropriate Command object.
+ */
+public class Parser {
+
+    /**
+     * Parses the user's input string and returns a Command object.
+     * If the command is not recognised, throws an UnknownCommandException.
+     *
+     * @param input The user's input string.
+     * @return A Command object corresponding to the user's input.
+     * @throws UnknownCommandException If the command is not recognised.
+     */
```

Comment:
Great job for leaving comments about the parameters of the methods and the exceptions it throws in your header!